src/Infrastructure/Filesystem/Filesystem.php

```diff
@@ -40,9 +40,13 @@ public function isWritable(string $path): bool
         return is_writable($path); // @codeCoverageIgnore
     }
 
-    public function remove(string $path): void
+    public function remove(string $path, ?int $timeout = 120): void
     {
         try {
+            // Symfony Filesystem doesn't have a builtin mechanism for setting a
+            // timeout, so we have to enforce it ourselves.
+            set_time_limit((int) $timeout);
```

Comment:
Not commit-blocking, but should we restore the previous time limit before this method finishes? According to https://www.php.net/manual/en/function.set-time-limit.php, we could get this from the `max_execution_time` php.ini value.

Reply:
Ah, good question. I'll merge the PR for now, and we can discuss this for follow-up at scrum. Thanks.